File: benchmark/comparison/tasks/17/solution.py
def list_zip(a, b):
    pairs = [f"Pair({x}, {y})" for x, y in zip(a, b)]
    return "[" + ", ".join(pairs) + "]"

print(list_zip([1, 2, 3], [4, 5, 6]))
print(list_zip([1, 2], [3]))
print(list_zip([], []))

File: benchmark/comparison/tasks/17/solution.ts
function list_zip(a: number[], b: number[]): string {
  const len = Math.min(a.length, b.length);
  const pairs: string[] = [];
  for (let i = 0; i < len; i++) {
    pairs.push(`Pair(${a[i]}, ${b[i]})`);
  }
  return "[" + pairs.join(", ") + "]";
}

console.log(list_zip([1, 2, 3], [4, 5, 6]));
console.log(list_zip([1, 2], [3]));
console.log(list_zip([], [1, 2]));
